controller/api/config.go

```diff
@@ -12,14 +12,42 @@ type ConfigUnset struct {
 
 // Config is the structure of an app's config.
 type Config struct {
-	Owner    string                 `json:"owner,omitempty"`
-	App      string                 `json:"app,omitempty"`
-	Values   map[string]interface{} `json:"values,omitempty"`
-	Memory   map[string]interface{} `json:"memory,omitempty"`
-	CPU      map[string]interface{} `json:"cpu,omitempty"`
-	Tags     map[string]interface{} `json:"tags,omitempty"`
-	Registry map[string]interface{} `json:"registry,omitempty"`
-	Created  string                 `json:"created,omitempty"`
-	Updated  string                 `json:"updated,omitempty"`
-	UUID     string                 `json:"uuid,omitempty"`
+	Owner        string                 `json:"owner,omitempty"`
+	App          string                 `json:"app,omitempty"`
+	Values       map[string]interface{} `json:"values,omitempty"`
+	Memory       map[string]interface{} `json:"memory,omitempty"`
+	CPU          map[string]interface{} `json:"cpu,omitempty"`
+	Healthchecks map[string]interface{} `json:"healthchecks,omitempty"`
+	Tags         map[string]interface{} `json:"tags,omitempty"`
+	Registry     map[string]interface{} `json:"registry,omitempty"`
+	Created      string                 `json:"created,omitempty"`
+	Updated      string                 `json:"updated,omitempty"`
+	UUID         string                 `json:"uuid,omitempty"`
+}
+
+type KVPair struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type ExecProbe struct {
+	Command []string `json:"command"`
+}
+
+type HTTPGetProbe struct {
+	Path        string   `json:"path"`
+	Port        string   `json:"port"`
+	HTTPHeaders []KVPair `json:"httpHeaders"`
```

Comment:
It is not old documentation:
http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_probe and http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_httpgetaction

Former should show you available types and the latter available httpget options

I found that doc page honestly much easier to read than the swagger UI :)

Just to reiterate: the Controller does not support host nor scheme and we have a want to add HttpHeaders but the others it does understand :)